src/app/clusters/time-synchronization-server/time-synchronization-cluster.cpp

```diff
@@ -136,151 +86,329 @@ static CHIP_ERROR UpdateUTCTime(uint64_t UTCTimeInChipEpochUs)
     return CHIP_NO_ERROR;
 }
 
-static bool emitDSTTableEmptyEvent(EndpointId ep)
-{
-    Events::DSTTableEmpty::Type event;
-    EventNumber eventNumber;
-
-    CHIP_ERROR error = LogEvent(event, ep, eventNumber);
+// -----------------------------------------------------------------------------
+// Event generation functions
 
-    if (CHIP_NO_ERROR != error)
+bool VerifyAndLogIfNoEventGenerator(EndpointId ep, DataModel::EventsGenerator * eventsGenerator)
+{
+    if (eventsGenerator == nullptr)
     {
-        ChipLogError(Zcl, "Unable to emit DSTTableEmpty event [ep=%d]", ep);
+        ChipLogError(Zcl, "Unable to emit event [ep=%d]. Event generator is null.", ep);
```

Comment:
for this class, is ep ever going to be something else then kRootEndpointId (i.e. 0)? If not, maybe we can save some flash by not trying to pass in the endpoint id and assume root everywhere.